src/java.net.http/share/classes/jdk/internal/net/http/Http1Exchange.java

```diff
@@ -208,6 +208,11 @@ private void error() {
             this.exchange = exchange;
         }
 
+        @Override
+        public void onSubscribed() {
+            exchange.registerResponseSubscriber(this);
```

Comment:
The `registerResponseSubscriber` in addition to registering it with the client will also set a local `responseSubscriber` member. Do you think we should `null` it out in the callbacks where we call the `unregisterResponseSubscriber`?

Reply:
Good question. But I believe not:  it wasn't needed before this change. The only place where we look at the subscriber is cancelImpl. Depending on the propagation of error conditions we might introduce a race if we did that - especially since we are unregistering the subscriber before propagating errors (if any).